## Implement Naïve Bayes method using scikit-learn library.

In [14]:
# importing all the required libraries
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.naive_bayes import GaussianNB
from sklearn import metrics
from sklearn.metrics import classification_report

## Read the glass dataset

In [15]:
# Read the dataset
data = pd.read_csv('/content/glass.csv')
X = data.drop('Type', axis=1)
y = data['Type'].values

## Split the data into Training and Test set

In [16]:
# Split the dataset into the Training set and Test set
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.25, random_state=0)

## Fit the Naïve Bayes model

In [17]:
# Fit Naive Bayes to the Training set
fitNB = GaussianNB()
fitNB.fit(X_train, y_train)

GaussianNB(priors=None, var_smoothing=1e-09)

## Evaluate the model on testing part using score

In [18]:
# Predicting the Test set results
y_pred = fitNB.predict(X_test)

# Calculating Accuracy
print('Accuracy of Naive Bayes model is: ', metrics.accuracy_score(y_test, y_pred))

Accuracy of Naive Bayes model is:  0.46296296296296297


## Use the model to predict the classes of the validation set & mean accuracy.

In [19]:
print(classification_report(y_test, y_pred))

              precision    recall  f1-score   support

           1       0.32      0.64      0.43        14
           2       0.45      0.21      0.29        24
           3       0.50      0.40      0.44         5
           5       0.00      0.00      0.00         2
           6       0.67      1.00      0.80         2
           7       1.00      1.00      1.00         7

    accuracy                           0.46        54
   macro avg       0.49      0.54      0.49        54
weighted avg       0.49      0.46      0.44        54



## From the Classification Report, Accuracy is around ~46% for Naïve Bayes Model.